### PR TITLE
fix!: helixCut is renamed to helixCutTolerance and moved to SeedFinderOptions

### PR DESCRIPTION
**Pull request by LuisFelipeCoelho**
Following up on PR #2472 I have:
- Changed the name `helixCut` to `helixCutTolerance`
- Moved `helixCutTolerance` to SeedFinderOptions
- Added it to Orthogonal config and SeedFinderFTFConfig

BREAKING CHANGE:

The option `helixCut` is renamed to `helixCutTolerance` and moved to `SeedFinderOptions`

### CODE REVIEW

File: Core/include/Acts/Seeding/SeedFinderConfig.hpp [hunk 1]
```diff
@@ -130,6 +130,11 @@ struct SeedFinderConfig {
   // default is 5%
   // TODO: necessary to make amount of material dependent on detector region?
   float radLengthPerSeed = 0.05;
+
+  // Parameter which can loosen the tolerance of the track seed to form to a
+  // helix, useful for (e.g.) misaligned seeding
```
Comment (AJPfleger) [inline Core/include/Acts/Seeding/SeedFinderConfig.hpp:135]:
```suggestion
  // Parameter which can loosen the tolerance of the track seed to form a
  // helix. This is useful for e.g. misaligned seeding.
```

File: Core/include/Acts/Seeding/SeedFinderFTFConfig.hpp [hunk 2]
```diff
@@ -42,6 +42,10 @@ struct SeedFinderFTFConfig {
   /// for load space points
   unsigned int maxSeedsPerSpM = 5;
 
+  // Parameter which can loosen the tolerance of the track seed to form to a
+  // helix, useful for (e.g.) misaligned seeding
```
Comment (AJPfleger) [inline Core/include/Acts/Seeding/SeedFinderFTFConfig.hpp:46]:
```suggestion
  // Parameter which can loosen the tolerance of the track seed to form a
  // helix. This is useful for e.g. misaligned seeding.
```

File: Core/include/Acts/Seeding/SeedFinderOrthogonalConfig.hpp [hunk 3]
```diff
@@ -101,6 +101,10 @@ struct SeedFinderOrthogonalConfig {
   // TODO: necessary to make amount of material dependent on detector region?
   float radLengthPerSeed = 0.05;
 
+  // Parameter which can loosen the tolerance of the track seed to form to a
+  // helix, useful for (e.g.) misaligned seeding
```
Comment (AJPfleger) [inline Core/include/Acts/Seeding/SeedFinderOrthogonalConfig.hpp:105]:
```suggestion
  // Parameter which can loosen the tolerance of the track seed to form a
  // helix. This is useful for e.g. misaligned seeding.
```

**Review by AJPfleger — DISMISSED** [review]

**Review by paulgessinger — COMMENTED** [review]
I don't see the removed config parameter in the diff. Where was this before? 
Also I suppose this is a breaking change?

**Comment by LuisFelipeCoelho** [comment]
> I don't see the removed config parameter in the diff. Where was this before? Also I suppose this is a breaking change?
> 
> EDIT: indeed the `helixCut` parameter is still there. Is this intentional?

Thank you @paulgessinger. I removed the parameter now.

Yes, I guess this is breaking change because of the rename.

**Comment by CarloVarni** [comment]
just to comment that in Athena we are currently not manually setting `helixCut` so this requires no change from the Athena side, despite  being a breaking change

**Review by paulgessinger — APPROVED** [review]

**Pull request merged**